src/targets/gpu/lowering.cpp

```diff
@@ -542,45 +542,20 @@ struct miopen_apply
                 new_inputs);
         });
 
-        apply_map.emplace("gpu::concat_past_present", [=](instruction_ref ins) {
-            return mod->replace_instruction(
-                ins,
-                make_op("gpu::precompile_op", {{"op", to_value(ins->get_operator())}}),
-                ins->inputs());
-        });
-
-        apply_map.emplace("gpu::compute_attention_probabilities", [=](instruction_ref ins) {
-            auto s          = ins->get_shape();
-            auto output     = insert_allocation(ins, s);
-            auto new_inputs = ins->inputs();
-            new_inputs.push_back(output);
-            return mod->replace_instruction(
-                ins,
-                make_op("gpu::precompile_op", {{"op", to_value(ins->get_operator())}}),
-                new_inputs);
-        });
-
-        apply_map.emplace("gpu::gqa_softmax", [=](instruction_ref ins) {
-            auto s      = ins->get_shape();
-            auto inputs = ins->inputs();
-
-            auto new_inputs = ins->inputs();
-            new_inputs.push_back(inputs.at(2));
-            return mod->replace_instruction(
-                ins,
-                make_op("gpu::precompile_op", {{"op", to_value(ins->get_operator())}}),
-                new_inputs);
+        apply_map.emplace("gpu::concat_past_present_k", [=](instruction_ref ins) {
+            return mod->replace_instruction(ins,
+                                            make_op("gpu::precompile_op",
+                                                    {{"op", to_value(ins->get_operator())},
+                                                     {"output_shape", to_value(ins->get_shape())}}),
+                                            ins->inputs());
```

Comment:
Is it necessary to do that manually? It should automatically wrap the op.

Reply:
Ah because there is no allocation inserted for this. Probably, should leave a comment for this.

Reply:
What should the comment say?